internal/manifests/collector/service.go

```diff
@@ -39,12 +39,17 @@ func (s ServiceType) String() string {
 }
 
 func HeadlessService(params manifests.Params) (*corev1.Service, error) {
+	// Use serviceName from spec if provided, otherwise default to <name>-headless for StatefulSets
```

Comment:
I'm wondering if this change can break somehow the `ServiceMonitor` integration.

Reply:
> @iblancasa don't we have an e2e test for this, in any case?
I don't think so.

I'm fine with the change because -after checking the code- it seems there will not be any kind of interactions...